### PR TITLE
Fix Recently Added Albums sort order (you will have to reset the Kodi database manually)

### LINKED ISSUE #793: Recently added music albums seem random
I am using Kodi 18.1 on Nvidia Shield TV. Applied direct paths.

Recently added music albums seem to show random albums (not the recently added ones from Plex).

Thanks for looking into this!

### PR DESCRIPTION
**Pull request by croneter**
- Fixes #793

### CODE REVIEW

**Comment by croneter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- resources/lib/backgroundthread.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/PlexKodiConnect/PlexKodiConnect/pullRequest?prid=4471247&bid=15097760)

File: resources/lib/library_sync/full_sync.py [hunk 1]
```diff
@@ -292,16 +293,22 @@ def threaded_get_iterators(self, kinds, queue, all_items=False):
...
+            (v.PLEX_TYPE_SHOW, v.PLEX_TYPE_SHOW, itemtypes.Show, False, Queue.Queue),
+            (v.PLEX_TYPE_SEASON, v.PLEX_TYPE_SHOW, itemtypes.Season, False, Queue.Queue),
+            (v.PLEX_TYPE_EPISODE, v.PLEX_TYPE_SHOW, itemtypes.Episode, False, Queue.Queue)
         ]
         if app.SYNC.enable_music:
             kinds.extend([
-                (v.PLEX_TYPE_ARTIST, v.PLEX_TYPE_ARTIST, itemtypes.Artist, False),
-                (v.PLEX_TYPE_ALBUM, v.PLEX_TYPE_ARTIST, itemtypes.Album, True),
+                (v.PLEX_TYPE_ARTIST, v.PLEX_TYPE_ARTIST, itemtypes.Artist, False, Queue.Queue),
+                (v.PLEX_TYPE_ALBUM, v.PLEX_TYPE_ARTIST, itemtypes.Album, True, backgroundthread.OrderedQueue),
```
Comment (croneter) [inline resources/lib/library_sync/full_sync.py]: Codacy Issue found: [Line too long (110/100)](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/backgroundthread.py [hunk 2]
```diff
@@ -140,6 +140,67 @@ def isSuspended(self):
...
+                    raise Queue.Empty
+            elif timeout is None:
+                while not self._qsize():
+                    self.not_empty.wait()
+                while self.queue[0][0] != self.smallest:
+                    self.not_next_item.wait()
+            elif timeout < 0:
+                raise ValueError("'timeout' must be a non-negative number")
+            else:
+                endtime = Queue._time() + timeout
```
Comment (croneter) [inline resources/lib/backgroundthread.py]: Codacy Issue found: [Access to a protected member _time of a client class](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/backgroundthread.py [hunk 3]
```diff
@@ -140,6 +140,67 @@ def isSuspended(self):
...
+                while not self._qsize():
+                    remaining = endtime - Queue._time()
+                    if remaining <= 0.0:
+                        raise Queue.Empty
+                    self.not_empty.wait(remaining)
+                while self.queue[0][0] != self.smallest:
+                    remaining = endtime - Queue._time()
```
Comment (croneter) [inline resources/lib/backgroundthread.py]: Codacy Issue found: [Access to a protected member _time of a client class](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/plex_functions.py [hunk 4]
```diff
@@ -629,34 +613,24 @@ def __init__(self, url, plex_type=None, last_viewed_at=None,
...
-                          self.url, self.args)
-                raise RuntimeError('Error while downloading chunks for %s'
-                                   % self.url)
-        else:
-            task = DownloadChunk(self.url,
-                                 deepcopy(self.args),  # Beware!
-                                 self.on_chunk_downloaded)
-            backgroundthread.BGThreader.addTask(task)
+    def set_xml(self, xml):
+        self.xml = xml
```
Comment (croneter) [inline resources/lib/plex_functions.py]: Codacy Issue found: [Attribute 'xml' defined outside __init__](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/backgroundthread.py [hunk 5]
```diff
@@ -140,6 +140,67 @@ def isSuspended(self):
         return self._suspended
 
 
+class OrderedQueue(Queue.PriorityQueue, object):
```
Comment (croneter) [inline resources/lib/backgroundthread.py]: Codacy Issue found: [Too few public methods (1/2)](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

**Comment by croneter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- resources/lib/backgroundthread.py  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/PlexKodiConnect/PlexKodiConnect/pullRequest?prid=4471247&bid=15097760)

File: resources/lib/backgroundthread.py (hunk 5 above)
Comment (croneter) [inline resources/lib/backgroundthread.py]: Codacy Issue found: [Too few public methods (1/2)](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/backgroundthread.py (hunk 3 above)
Comment (croneter) [inline resources/lib/backgroundthread.py]: Codacy Issue found: [Access to a protected member _time of a client class](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/backgroundthread.py (hunk 2 above)
Comment (croneter) [inline resources/lib/backgroundthread.py]: Codacy Issue found: [Access to a protected member _time of a client class](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

File: resources/lib/library_sync/full_sync.py [hunk 6]
```diff
@@ -292,16 +294,22 @@ def threaded_get_iterators(self, kinds, queue, all_items=False):
...
```
Comment (croneter) [inline resources/lib/library_sync/full_sync.py]: Codacy Issue found: [Line too long (110/100)](https://app.codacy.com/app/croneter/PlexKodiConnect/pullRequest?prid=4471247)

**Comment by SwangDemon** [comment]
Hi, my apologies if I'm missing something, but I am still experiencing this issue, and the deleted comments in this thread make this conversation a little difficult to follow. My 'recently added albums' widget is essentially in reverse order. I read an explanation as to why this happens in another thread, which makes sense, but I can't seem to rectify the issue. I have tried reloading the database and editing the advancedsettings.xml, but I am still at a loss. Everything else is fine on the plex end and otherwise it seems like a great addon, so thank you! In short, as far as I can tell the issue is fixed, but what exactly is the fix?

**Pull request merged**